plugins/task/deleteactionlogs/services/provider.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Task.deletelogs
+ *
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Database\DatabaseInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Task\DeleteActionLogs\Extension\DeleteActionLogs;
+
+return new class () implements ServiceProviderInterface {
+    /**
+     * Registers the service provider with a DI container.
+     *
+     * @param   Container  $container  The DI container.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function register(Container $container)
+    {
+        $container->set(
+            PluginInterface::class,
+            function (Container $container) {
+                $dispatcher = $container->get(DispatcherInterface::class);
+
+                $plugin = new DeleteActionLogs(
+                    $dispatcher,
+                    (array) PluginHelper::getPlugin('task', 'deleteactionlogs')
```

Comment:
```suggestion
                $plugin = new DeleteActionLogs(
                    $container->get(DispatcherInterface::class),
                    (array) PluginHelper::getPlugin('task', 'deleteactionlogs')
```
See #40746 - thanks @heelc29 for the hint in the other PR.